sklearn/isotonic.py

```diff
@@ -320,6 +323,19 @@ def fit(self, X, y, sample_weight=None):
         self.X_min_ = np.min(self.X_)
         self.X_max_ = np.max(self.X_)
 
+        if fast_predict:
+            # Remove unnecessary points for faster prediction
+            fitted_len = len(self.y_)
+            keep_data = np.ones((fitted_len,), dtype=bool)
+            # Aside from the 1st and last point, remove points whose y values
+            # are equal to both the point before and the point after it.
+            keep_data[1:fitted_len-1] = np.logical_or(
```

Comment:
You can probably use `keep_data[1:-1]` here (assuming np arrays behave like Python lists with negative indexing?)

Also in the slices below.